### PR TITLE
pricing api updated to v2

### PR DESCRIPTION
**Pull request by mikluke**

### CODE REVIEW

File: prices_voice.go [hunk 1]
```diff
@@ -20,11 +20,20 @@ type NumberVoicePriceService struct {
...
 
+func numberVoicePricePart(number, originationNumber string) string {
+	res := number
+	if originationNumber != "" {
+		res += "?OriginationNumber=" + originationNumber
+	}
+	return res
+}
+
 type PrefixPrice struct {
```
Comment (kevinburke) [inline prices_voice.go:24]: This change is going to break everyone who currently relies on this code. How should we handle this?

File: prices_voice.go [hunk 2]
```diff
@@ -48,13 +58,15 @@ type VoicePrice struct {
...
-	InboundCallPrice  InboundPrice      `json:"inbound_call_price"`
-	OutboundCallPrice OutboundCallPrice `json:"outbound_call_price"`
-	PriceUnit         string            `json:"price_unit"`
-	URL               string            `json:"url"`
+	Country            string              `json:"country"`
+	IsoCountry         string              `json:"iso_country"`
+	DestinationNumber  string              `json:"destination_number"`
+	OriginationNumber  string              `json:"origination_number"`
+	InboundCallPrice   InboundPrice        `json:"inbound_call_price"`
+	OutboundCallPrices []OutboundCallPrice `json:"outbound_call_prices"`
```
Comment (kevinburke) [inline prices_voice.go:59]: Ditto here, this used to be a single price and is now multiple.

File: prices_voice.go [hunk 3]
```diff
@@ -64,10 +76,10 @@ func (cvps *CountryVoicePriceService) Get(ctx context.Context, isoCountry string
 	return voicePrice, err
 }
 
-// returns the call price by number
-func (nvps *NumberVoicePriceService) Get(ctx context.Context, number string) (*VoiceNumberPrice, error) {
+// returns the call price by number or numbers pair
```
Comment (kevinburke) [inline prices_voice.go:74]: Can we add a link to the API docs here? Apologies, as I should have done this the first time.

File: prices_voice.go (hunk 1 above)
Comment (mikluke) [reply prices_voice.go:24]: as a variant we can make v2 branch for v2 api version

or I can make function with and without origination number, but response changed anyway

Comment (kevinburke) [reply prices_voice.go:24]: Huh. TBH I haven't been too happy with that format. I think maybe we should just upgrade the library version to 2.0.

Can you update CHANGELOG.md with the required changes if you have code that's working with v1, to get it working again with v2?

File: prices_voice.go (hunk 3 above)
Comment (mikluke) [reply prices_voice.go:74]: done

File: prices_voice.go (hunk 1 above)
Comment (mikluke) [reply prices_voice.go:24]: done

File: prices_voice.go [hunk 4]
```diff
@@ -48,26 +58,32 @@ type VoicePrice struct {
...
 	voicePrice := new(VoicePrice)
 	err := cvps.client.GetResource(ctx, voicePathPart+"/Countries", isoCountry, voicePrice)
 	return voicePrice, err
 }
 
-// returns the call price by number
-func (nvps *NumberVoicePriceService) Get(ctx context.Context, number string) (*VoiceNumberPrice, error) {
+// https://www.twilio.com/docs/voice/pricing#list-uri-by-origination-number
+// returns the call price by number or numbers pair
+func (nvps *NumberVoicePriceService) Get(ctx context.Context, destinationNumber string,
```
Comment (kevinburke) [inline prices_voice.go]: I don't think this is quite right. The URL structure is:

```
https://pricing.twilio.com/v2/Voice/Numbers/{Destination Number}?OriginationNumber={Origination Number}
```

Elsewhere in the API, we handle parameters that are part of the path by adding them as required fields as we've done here, but we handle query parameters as a `url.Values`. So I think the API should be:

```
Get(ctx context.Context, destinationNumber string, data url.Values)
```

We can indicate either in the docstring or an example (or both) that the one valid value for `data` currently is `OriginationNumber`. But I think it is a mistake to include it as part of the method declaration - what if Twilio adds another parameter or changes its type? The url.Values method is future proof, hand coding arguments is not.

Comment (kevinburke) [reply prices_voice.go]: As a side effect this means you can get rid of `numberVoicePricePart` I think and just pass through a url.Values

Comment (mikluke) [reply prices_voice.go]: done

File: prices_voice.go [hunk 5]
```diff
@@ -48,26 +52,31 @@ type VoicePrice struct {
...
 
-// returns the call price by number
-func (nvps *NumberVoicePriceService) Get(ctx context.Context, number string) (*VoiceNumberPrice, error) {
+// https://www.twilio.com/docs/voice/pricing#list-uri-by-origination-number
+// returns the call price by number or numbers pair
+func (nvps *NumberVoicePriceService) Get(ctx context.Context, destinationNumber string, data url.Values) (*VoiceNumberPrice, error) {
 	voiceNumPrice := new(VoiceNumberPrice)
-	err := nvps.client.GetResource(ctx, voicePathPart+"/Numbers", number, voiceNumPrice)
+	pathPart := strings.Join([]string{voicePathPart, "Numbers", destinationNumber}, "/")
+	err := nvps.client.MakeRequest(ctx, http.MethodGet, pathPart, data, voiceNumPrice)
```
Comment (kevinburke) [inline prices_voice.go]: ah, you can use ListResource here I think

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,19 @@
 # Changes
 
+## 1.9
```
Comment (kevinburke) [inline CHANGELOG.md]: Change to 2.0

**Review by kevinburke — COMMENTED** [review]
We are almost there!

File: CHANGELOG.md (hunk 6 above)
Comment (mikluke) [reply CHANGELOG.md]: done

File: prices_voice.go (hunk 5 above)
Comment (mikluke) [reply prices_voice.go]: I don't think so
There is not listable resource, I think new ParametrizedGet method will be more obviously.

anyway, done

**Comment by kevinburke** [comment]
Merged as d8ca0d2a70c75391f618c0d0690c8b91f44b6c73, thanks!

**Pull request closed without merging**